### PR TITLE
docs(components): docs on component internals

### PR DESCRIPTION
**Pull request by asudoh**
### Added

Deep-dive documentation on Carbon component mixins.

## Testing / Reviewing

Reviewing should ensure the newly added document will make the internals of Carbon component mixins easy to understand.

### CODE REVIEW

File: src/globals/js/mixins/README.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
+# Carbon component mixins
+
+Carbon component mixins, based on [Subclass Factory Pattern](https://github.com/justinfagnani/proposal-mixins#subclass-factory-pattern), provides the basis for Carbon component classes by allowing component implementation to compose small pieces of functionalities to base them on, instead of introducing "fat base class".
+
+* [Component lifecycle](#component-lifecycle-create-componentjs) ([`create-component.js`](./create-component.js))
```
Comment (joshblack) [inline src/globals/js/mixins/README.md]: You could use [`doctoc`](https://github.com/thlorenz/doctoc) here to generate the table of contents if you wanted 😄 

Used in DataTable in the README: https://raw.githubusercontent.com/carbon-design-system/carbon-components-react/master/src/components/DataTable/README.md

The snippet itself just looks like:

```md
<!-- To run doctoc, just do `npx doctoc README.md` in this directory! -->

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->


- [Installation](#installation)
- [Usage](#usage)
- [Props](#props)
  - [`rows`](#rows)
  - [`headers`](#headers)
  - [`sortRow`](#sortrow)
  - [`filterRows`](#filterrows)
  - [`locale`](#locale)
- [Render Prop Function](#render-prop-function)
  - [Prop Getters](#prop-getters)
  - [Actions](#actions)
  - [State](#state)
  - [Props](#props-1)
- [Use-cases](#use-cases)
  - [Sorting](#sorting)
    - [Programmatic sorting](#programmatic-sorting)
    - [Custom sorting](#custom-sorting)
  - [Expansion](#expansion)
    - [Programmatic expansion](#programmatic-expansion)
  - [Selection](#selection)
    - [Programmatic selection](#programmatic-selection)
  - [Filtering](#filtering)
  - [Batch Actions](#batch-actions)
- [Attribution](#attribution)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->
```

**Comment by joshblack** [comment]
I think this is great 💯 Thanks so much for doing it!

Is there any chance, in the future, to also use the JSDoc comments to auto-generate some useful stuff to embed here as well? Or maybe in a separate docs page?

**Comment by asudoh** [comment]
Thanks @joshblack for suggesting TOC thing, will try. Wrt JSDoc, currently people can see it by:

```javascript
> gulp jsdoc
> open docs/js/index.html
```

Deploying it is one of long-standing back-log items. 😅

File: src/globals/js/mixins/README.md (hunk 1 above)
Comment (asudoh) [reply src/globals/js/mixins/README.md]: Done, thanks @joshblack for suggesting!

**Review by marijohannessen — APPROVED** [review]
Awesome! 👍 ✅

**Comment by asudoh** [comment]
Updated TOC with automatically generated version. Thanks again @joshblack for your suggestion!

**Review by tw15egan — APPROVED** [review]
This is awesome, thanks @asudoh! 👍 ✅

**Pull request merged**